File: Fib.java
import java.util.*;

public class Fib {
  static long fib(long n) {
    if (n <= 1)
      return 1;
    else
      return fib(n - 1) + fib(n - 2);
  }

  public static void main(String[] args) {
    System.out.print(fib(46));
  }
}

File: fib.zig
const std = @import("std");

fn fib(x: u64) u64 {
    if (x <= 1) return x;
    return fib(x - 1) + fib(x - 2);
}

pub fn main() void {
    std.debug.warn("{}", fib(46));
}
